### PR TITLE
fix(discover): Empty top N chart should have x axis

### PR DESCRIPTION
**Pull request by Zylphrex**
The events stats endpoint in top events mode returns an empty dictionary when
there are no top events. This results in a blank graph with no x-axis values. This
change falls back to returning a single zero-filled series in order to preserve
the x-axis.

# Before

image

# After

image

### CODE REVIEW

**Review by dashed — APPROVED** [review]

File: tests/snuba/api/endpoints/test_organization_events_stats.py [hunk 1]
```diff
@@ -725,6 +725,31 @@ def setUp(self):
...
+                    "topEvents": 5,
+                },
+                format="json",
+            )
+
+        data = response.data["data"]
+        assert response.status_code == 200, response.content
+        # When there are no top events, we do not return an empty dict.
+        # Instead, we return a single zero-filled series for an empty graph.
+        assert [attrs for time, attrs in data] == [[{"count": 0}], [{"count": 0}]]
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_organization_events_stats.py:751]: Should the response shape stay the same as if a topN hit was found? You could use '' as the series key.

Comment (Zylphrex) [reply tests/snuba/api/endpoints/test_organization_events_stats.py:751]: Using the empty string as the series key will keep the response shape the same but there are some drawbacks.

The empty string is technically a valid value, and thus a potential series name, so we won't have any way of distinguishing between the two. This means that we can't label the tooltip properly for a series with the empty string as the name vs a no results series.

**Review by markstory — APPROVED** [review]

File: src/sentry/snuba/discover.py [hunk 2]
```diff
@@ -681,6 +682,14 @@ def top_events_timeseries(
             referrer=referrer,
         )
 
+    if not allow_empty and not result.get("data", []):
```
Comment (wmak) [inline src/sentry/snuba/discover.py:685]: nit: Should we use `len` here instead of treating `[]` as falsey

Comment (Zylphrex) [reply src/sentry/snuba/discover.py]: fixed in 10f9070

**Review by wmak — APPROVED** [review]

**Pull request merged**